### ISSUE TITLE
Mapstruct: @InheritInverseConfiguration --> There is no suitable result type constructor for reversing this method

### ISSUE THREAD

**Issue by mavarazo** [mapstruct/mapstruct#1283]
I try to use mapstruct feature @InheritInverseConfiguration. But it gives me the following error: There is no suitable result type constructor for reversing this method. But the generated Impl look pretty good.
```
@Mapper(uses = {
  BOIdMapper.class,
  DateMapper.class,
  CarVOFactory.class,
})
public interface CarMapper {
   CarMapper INSTANCE = Mappers.getMapper(CarMapper .class);

   @Mappings({
     @Mapping(source = "VOKey", target = "id"),
     @Mapping(source = "gueltAb", target = "gueltigAb")
   })
   Car map(CarVO carVO);

   @InheritInverseConfiguration
   CarVO map(Car car);

   public class CarVOFactory {
      public CarVO createCarVO() {
         return VOFactory.createVO(CarVO.class);
      }
   }
}
```

This is the POJO Car
```
public class Car {

    private String id;
    private LocalDate gueltigAb;
    ...
}
```
This is the other Car-Class:
```
public interface CarVO extends ... {
    void setVOKey (VOKey<CAR> voKey);
}
```
And this is the generated Impl:
```
@Generated(
    value = "org.mapstruct.ap.MappingProcessor",
    date = "2017-08-29T08:55:10+0200",
    comments = "version: 1.1.0.Final, compiler: Eclipse JDT (IDE) 1.2.100.v20160418-1457, environment: Java 1.8.0_102 (Oracle Corporation)"
)

public class CarMapperImpl implements CarMapper {
    private final BOIdMapper bOIdMapper = new BOIdMapper();
    private final DateMapper dateMapper = new DateMapper();
    private final CarVOFactory CarVOFactory = new CarVOFactory();

    @Override
    public Car map(CarVO CarVO) {
        if ( CarVO == null ) {
            return null;
        }
        Car Car = new Car();
        Car.setGueltigAb( dateMapper.asLocalDate( CarVO.getGueltAb() ) );
        Car.setId( bOIdMapper.asId( CarVO.getVOKey() ) );
        return Car;
   }
    @Override
    public CarVO map(Car Car) {
        if ( Car == null ) {
            return null;
        }
        CarVO CarVO = CarVOFactory.createCarVO();
        return CarVO;
   }

```

[Linked](https://stackoverflow.com/questions/45912643/mapstruct-inheritinverseconfiguration-there-is-no-suitable-result-type-con)

**Comment by filiphr** [comment]
Thanks for reporting this @mavarazo. I'll have a look at it and a fix will probably be part of `1.2.0.Final`.

Btw. The generated mapper is wrong, the reverse properties are not mapped in the reverse method.

**Comment by filiphr** [comment]
Functionality for this has been implemented in #838. However, it seems that there are more cases when that should pass 😄.

@sjaakd Why do we need a special case for that, why not leave the error handling to the `BeanMappingMethod`? If we leave it there then it should work correctly right?

**Comment by filiphr** [comment]
@mavarazo This should be fixed once we merge #1287 in case you have the chance can you please try out my PR?

**Comment by filiphr** [comment]
@mavarazo, this has been merged and fixed into master. The final release with the fix should follow soon
